File: python/foo.py
#!/usr/bin/env python3

l, d, kv = [1,2,3], (1,2,3), {1:2,3:4}
s = 'hello'

print("%s, %d%d%d%d" % (s, l[0], d[1], 3, kv[3]))
print("{}, {}{}{}{}".format(s, l[0], d[1], 3, kv[3]))
print(f"{s}, {l[0]}{d[1]}3{kv[3]}")

File: gap/foo.pl
#!/usr/bin/env perl

$post = 0;
foreach $line (`cat c`) {
    $line =~ s/\r//;
    $line =~ s/\n//;
    
    $now = $line;
    $gap = $now - $post;
    printf("$now\t$gap \n");
    
    $post = $now;
}
